src/pprof/ddprof_pprof.cc

```diff
@@ -136,17 +123,16 @@ static void write_line(const ddprof::Symbol &symbol,
 // Assumption of API is that sample is valid in a single type
 DDRes pprof_aggregate(const UnwindOutput *uw_output,
                       const SymbolHdr *symbol_hdr, uint64_t value,
-                      int watcher_idx, DDProfPProf *pprof) {
+                      const PerfWatcher *watcher, DDProfPProf *pprof) {
 
   const ddprof::SymbolTable &symbol_table = symbol_hdr->_symbol_table;
   const ddprof::MapInfoTable &mapinfo_table = symbol_hdr->_mapinfo_table;
   ddprof_ffi_Profile *profile = pprof->_profile;
 
-  int64_t values[MAX_TYPE_WATCHER + 1] = {0};
-  // Counted a single time
-  values[0] = 1;
-  // Add a value to the watcher we are sampling, leave others zeroed
-  values[watcher_idx + 1] = value;
+  int64_t values[DDPROF_PWT_LENGTH] = {0};
+  values[watcher->profile_id] = value;
+  if (is_countable_type(watcher->profile_id))
+    values[watcher->count_id] = 1;
```

Comment:
`count_id` is not initialized

Reply:
Indeed, I meant to factor it away entirely.  Good catch--thanks!

Reply:
Fixed in latest commit